From 7f9ca362063735577b21aa72e9620d0f25bf66c6 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Fri, 09 Apr 2021 11:25:09 +0300
Subject: [PATCH] [SW-36011] gaudi: use struct for nic enable

Instead of using bitmask only for nic enable, use
of struct hl_nic_en which contains both bitmask
and validity, so that validity is used to ensure
that bitmask was written by KVM. If validity is 1,
it means that bitmask is valid. This flag is set
when nic enable fit is received, and is only used
in preboot internally.

Change-Id: I96d6d9af45b1920ae93f406422f36d230ea2aab0
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 9c22025..1614d7b 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -465,7 +465,7 @@
 	return (u32)mmPCIE_DBI_BASE & 0xFFFFFF;
 }
 
-static u32 nic_en;
+static struct hl_nic_en nic_en;
 
 #define NIC_PHY_RX_CFG_OFFSET	0x10
 #define NIC_PORT_EN		0
@@ -492,8 +492,9 @@
 
 void preboot_nic_en(u32 nic_en_)
 {
-	nic_en = nic_en_;
-	enable_nic_regs(nic_en);
-	preboot_puts_hex("nic_en ", nic_en, true);
+	nic_en.is_valid = true;
+	nic_en.bitmask = nic_en_;
+	enable_nic_regs(nic_en_);
+	preboot_puts_hex("nic_en ", nic_en_, true);
 	boot_info_set(HL_BOOT_INFO_NIC_EN, &nic_en, sizeof(nic_en));
 }
diff --git a/include/preboot.h b/include/preboot.h
index 5424e9f..7603a0e 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -73,6 +73,11 @@
 	PREBOOT_EFUSE_VER,
 };
 
+struct hl_nic_en {
+	bool is_valid;
+	uint32_t bitmask;
+};
+
 extern uintptr_t _start;	/* Start of ppboot/preboot region */
 extern uintptr_t __stack_top;	/* End of ppboot/preboot region */
 
